internal/compiler/llr/lower_to_item_tree.rs

```diff
@@ -647,9 +655,23 @@ fn lower_popup_component(
     component: &Rc<Component>,
     ctx: &ExpressionContext,
     compiler_config: &CompilerConfiguration,
-) -> ItemTree {
-    let sc = lower_sub_component(component, ctx.state, Some(ctx), compiler_config);
-    ItemTree {
+    x: &NamedReference,
+    y: &NamedReference,
+) -> llr_PopupWindow {
+    let mut x_prop = None;
+    let mut y_prop = None;
+
+    let sc = lower_sub_component(
+        component,
+        ctx.state,
+        Some(ctx),
+        compiler_config,
+        Some(Box::new(|ctx| {
+            x_prop = Some(ctx.map_property_reference(x));
+            y_prop = Some(ctx.map_property_reference(y));
+        })),
```

Comment:
I'm not a fan of this callback.
Look at what we do for the listview: `sc.mapping.map_property_reference(&geom.y, ctx.state)`  doesn't it work in this case?

Reply:
I also dislike it :(. You're right about `sc.mapping` - that should work, since this can't be a reference to a parent. Thanks!